### PR TITLE
docs: change gatsby-image references to gatsby-plugin-image

### PR DESCRIPTION
**Pull request by wardpeet**
### Documentation

Update a few docs that reference gatsby-image and move them to gatsby-plugin-image

## Related Issues

Part of [ch32168]

### CODE REVIEW

File: docs/docs/conceptual/security-in-gatsby.md [hunk 1]
```diff
@@ -149,7 +149,7 @@ Sometimes in your Gatsby website, you will need display sensitive data or handle
 Content Security Policy is a security layer added in web applications to detect and prevent attacks, e.g. the XSS attack mentioned above.
 
 To add it to your Gatsby website, add [gatsby-plugin-csp](/plugins/gatsby-plugin-csp/) to your `gatsby-config.js` with the desired configuration. Note that
-currently there is a [compatibility issue](https://github.com/gatsbyjs/gatsby/issues/10890) between [gatsby-plugin-csp](/plugins/gatsby-plugin-csp/) and other plugins that generate hashes in inline styles, including [gatsby-image](/plugins/gatsby-image).
+currently there is a [compatibility issue](https://github.com/gatsbyjs/gatsby/issues/10890) between [gatsby-plugin-csp](/plugins/gatsby-plugin-csp/) and other plugins that generate hashes in inline styles, including [gatsby-plugin-image](/plugins/gatsby-plugin-image).
```
Comment (fk) [inline docs/docs/conceptual/security-in-gatsby.md:152]: :doge: question — the bit about compatibility and hash generation is true for `gatsby-plugin-image`, too?

File: docs/docs/creating-a-source-plugin.md [hunk 2]
```diff
@@ -413,7 +413,9 @@ query {
     id
     remoteImage {
       childImageSharp {
-        # fluid or fixed fields for optimized images
+        gatsbyImageData(
+          # fluid or fixed fields for optimized images
```
Comment (fk) [inline docs/docs/creating-a-source-plugin.md:417]: Should this be

```
# arguments for optimized images
```

now?

(Sorry, not trying to be nit picky, learning about `gatsby-plugin-image` still :-/)

File: docs/docs/conceptual/security-in-gatsby.md (hunk 1 above)
Comment (wardpeet) [reply docs/docs/conceptual/security-in-gatsby.md:152]: yes, we add some inline styles

File: docs/docs/sourcing-from-sanity.md [hunk 3]
```diff
@@ -65,20 +65,20 @@ Gatsby cannot know about the types and fields without having documents of the gi
...
 ```jsx
 import React from "react"
-import Img from "gatsby-image"
+import { GatsbyImage } from "gatsby-plugin-image"
 
 const Person = ({ data }) => (
   <article>
     <h2>{data.sanityPerson.name}</h2>
-    <Img fluid={data.sanityPerson.profileImage.asset.fluid} />
+    <GatsbyImage fluid={data.sanityPerson.profileImage.asset.gatsbyImageData} />
```
Comment (fk) [inline docs/docs/sourcing-from-sanity.md:81]: > `fluid` images no longer exist

https://www.gatsbyjs.com/docs/reference/release-notes/image-migration-guide/#breaking-changes :/

File: docs/docs/creating-a-source-plugin.md (hunk 2 above)
Comment (wardpeet) [reply docs/docs/creating-a-source-plugin.md:417]: That would work :D

File: docs/docs/conceptual/security-in-gatsby.md (hunk 1 above)
Comment (fk) [reply docs/docs/conceptual/security-in-gatsby.md:152]: 👍

File: docs/docs/sourcing-from-sanity.md [hunk 4]
```diff
@@ -65,20 +65,20 @@ Gatsby cannot know about the types and fields without having documents of the gi
 
 ## Using images
 
-Image fields will have the image URL available under the `field.asset.url` key, but you can also use [gatsby-image](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-image) for a smooth experience. It's a React component that enables responsive images and advanced image loading techniques. It works great with this source plugin, without requiring any additional build steps.
+Image fields will have the image URL available under the `field.asset.url` key, but you can also use [gatsby-plugin-image](/plugins/gatsby-plugin-image/) for a smooth experience. It's a React component that enables responsive images and advanced image loading techniques. It works great with this source plugin, without requiring any additional build steps.
 
 There are two types of responsive images supported; _fixed_ and _fluid_. To decide between the two, ask yourself: "do I know the exact size this image will be?" If yes, you'll want to use _fixed_. If no and its width and/or height need to vary depending on the size of the screen, then you'll want to use _fluid_.
```
Comment (fk) [inline docs/docs/sourcing-from-sanity.md:70]: Needs an update, too (`fluid` -> `fullWidth`). Unfortunately I don't have enough context (yet) to make a suggestion for the `decide between the two` part.

**Comment by bradledford** [comment]
This is a quite old PR with seemingly just a single comment blocking progress https://github.com/gatsbyjs/gatsby/pull/31727/files#r650839982.

Anything we can do to get this merged and off our plates?

**Pull request merged**